packages/material-ui-system/src/createStyled.d.ts

```diff
@@ -77,17 +77,7 @@ export type Overwrapped<T, U> = Pick<T, Extract<keyof T, keyof U>>;
 
 export interface StyledComponent<InnerProps, StyleProps, Theme extends object>
   extends React.FunctionComponent<InnerProps & StyleProps & { theme?: Theme }>,
-    ComponentSelector {
-  /**
-   * @desc this method is type-unsafe
-   */
-  withComponent<NewTag extends keyof JSX.IntrinsicElements>(
-    tag: NewTag,
-  ): StyledComponent<JSX.IntrinsicElements[NewTag], StyleProps, Theme>;
-  withComponent<Tag extends React.JSXElementConstructor<any>>(
-    tag: Tag,
-  ): StyledComponent<PropsOf<Tag>, StyleProps, Theme>;
-}
+    ComponentSelector {}
```

Comment:
Do we want to push it the extra mile with the tradeoff of @dimitropoulos in https://github.com/mui-org/material-ui/issues/20012#issuecomment-899578418

```suggestion
    ComponentSelector {
      /**
       * Use the as prop instead. It's not style destructive when using multiple layers of styled() calls
       */
      withComponent(tag: any): never
    }
```

?

Reply:
Actually, it doesn't work. It would silence the missing support in TypeScript.